search/services/index.py

```diff
@@ -0,0 +1,785 @@
+"""
```

Comment:
This file is too big. Seems to include sessions to ES, query generation, searching, indexing, Error classes, searching etc.

Refactor this into more focused units and combine with composition.

Reply:
Agreed that it's due for some refactoring. Should be addressed in https://github.com/cul-it/arxiv-search/issues/112 between beta and final.